src/array_partition.jl

```diff
@@ -24,31 +24,33 @@ function Base.ones(A::ArrayPartition)
   B
 end
 
-Base.:+(A::ArrayPartition, B::ArrayPartition) = ArrayPartition((x .+ y for (x,y) in zip(A.x,B.x))...)
+Base.:+(A::ArrayPartition, B::ArrayPartition) =
+    ArrayPartition((x .+ y for (x,y) in zip(A.x,B.x))...)
 Base.:+(A::Number, B::ArrayPartition) = ArrayPartition((A .+ x for x in B.x)...)
 Base.:+(A::ArrayPartition, B::Number) = ArrayPartition((B .+ x for x in A.x)...)
-Base.:-(A::ArrayPartition, B::ArrayPartition) = ArrayPartition((x .- y for (x,y) in zip(A.x,B.x))...)
+Base.:-(A::ArrayPartition, B::ArrayPartition) =
+    ArrayPartition((x .- y for (x,y) in zip(A.x,B.x))...)
 Base.:-(A::Number, B::ArrayPartition) = ArrayPartition((A .- x for x in B.x)...)
 Base.:-(A::ArrayPartition, B::Number) = ArrayPartition((x .- B for x in A.x)...)
 Base.:*(A::Number, B::ArrayPartition) = ArrayPartition((A .* x for x in B.x)...)
 Base.:*(A::ArrayPartition, B::Number) = ArrayPartition((x .* B for x in A.x)...)
 Base.:/(A::ArrayPartition, B::Number) = ArrayPartition((x ./ B for x in A.x)...)
 Base.:\(A::Number, B::ArrayPartition) = ArrayPartition((x ./ A for x in B.x)...)
 
-if VERSION < v"0.6-"
-  Base.:.+(A::ArrayPartition, B::ArrayPartition) = ArrayPartition((x .+ y for (x,y) in zip(A.x,B.x))...)
-  Base.:.+(A::Number, B::ArrayPartition) = ArrayPartition((A .+ x for x in B.x)...)
-  Base.:.+(A::ArrayPartition, B::Number) = ArrayPartition((B .+ x for x in A.x)...)
-  Base.:.-(A::ArrayPartition, B::ArrayPartition) = ArrayPartition((x .- y for (x,y) in zip(A.x,B.x))...)
-  Base.:.-(A::Number, B::ArrayPartition) = ArrayPartition((A .- x for x in B.x)...)
-  Base.:.-(A::ArrayPartition, B::Number) = ArrayPartition((x .- B for x in A.x)...)
-  Base.:.*(A::ArrayPartition, B::ArrayPartition) = ArrayPartition((x .* y for (x,y) in zip(A.x,B.x))...)
-  Base.:.*(A::Number, B::ArrayPartition) = ArrayPartition((A .* x for x in B.x)...)
-  Base.:.*(A::ArrayPartition, B::Number) = ArrayPartition((x .* B for x in A.x)...)
-  Base.:./(A::ArrayPartition, B::ArrayPartition) = ArrayPartition((x ./ y for (x,y) in zip(A.x,B.x))...)
-  Base.:./(A::ArrayPartition, B::Number) = ArrayPartition((x ./ B for x in A.x)...)
-  Base.:.\(A::Number, B::ArrayPartition) = ArrayPartition((x ./ A for x in B.x)...)
-end
+VERSION < v"0.6.0-dev.1614" && include_string("""
```

Comment:
Specifically, I changed the code to:

```julia
@static if VERSION < v"0.6-"
  Base.:.+(A::ArrayPartition, B::ArrayPartition) = ArrayPartition((x .+ y for (x,y) in zip(A.x,B.x))...)
  <other methods omitted>
end
```

and got no deprecation warnings after changing the code and re-importing it in a new Julia session.